### ISSUE TITLE
ERROR:  relation "bucardo.bucardo_truncate_trigger" does not exist

### ISSUE THREAD

**Issue by mailbaoer** [bucardo/bucardo#160]
I use bucardo  to export data from aws rds, when I run bucardo start，I saw the log 
`
23691) [Mon Dec 11 16:03:24 2017] MCP Starting check_sync_health
(28658) [Mon Dec 11 16:03:35 2017] KID (the_sync) New kid, sync "the_sync" alive=1 Parent=23731 PID=28658 kicked=1
(28658) [Mon Dec 11 16:03:35 2017] KID (the_sync) Kid has died, error is: DBD::Pg::st execute failed: ERROR:  relation "bucardo.bucardo_truncate_trigger" does not exist LINE 1: ...ident(tname), MAX(EXTRACT(epoch FROM cdate)) FROM bucardo.bu...                                                              ^ at Bucardo.pm line 3303. Line: 5037 Main DB state: ? Error: none DB dest_db state: ? Error: none DB source_db state: 42P01 Error: 7
(28658) [Mon Dec 11 16:03:35 2017] KID (the_sync) Kid 28658 exiting at cleanup_kid. Sync "the_sync" Reason: DBD::Pg::st execute failed: ERROR:  relation "bucardo.bucardo_truncate_trigger" does not exist LINE 1: ...ident(tname), MAX(EXTRACT(epoch FROM cdate)) FROM bucardo.bu...                                                              ^ at Bucardo.pm line 3303. Line: 5037 Main DB state: ? Error: none DB dest_db state: ? Error: none DB source_db state: 42P01 Error: 7
`

It seems that I didn't create the bucardo_truncate_trigger table, but when I created it , It didn't work, What can I do next?

**Comment by makeittotop** [comment]
I've also experienced the same issue today while using bucardo on premise. Any help or pointers would be greatly appreciated.

**Comment by saifulmuhajir** [comment]
`bucardo.bucardo_truncate_trigger` should be there when sync is created. If it is not, most probably the installation was failed at some point.

You can find the information about the relation [here.](https://github.com/bucardo/bucardo/blob/cf5988d735e3213982b68dcea6ff60ab07535e88/bucardo.schema#L1989)

**Comment by danieltanfh95** [comment]
still having this problem. 

Solution: 

1. Create the table manually using
```
CREATE TABLE bucardo.bucardo_truncate_trigger (
                        tablename   OID         NOT NULL,
                        sname       TEXT        NOT NULL,
                        tname       TEXT        NOT NULL,
                        sync        TEXT        NOT NULL,
                        replicated  TIMESTAMPTZ     NULL,
                        cdate       TIMESTAMPTZ NOT NULL DEFAULT now()
);
```

2. Restart bucardo.
3. After adding a new table (not the table above), update sync with all relgroups and herds. Restart bucardo again.

**Comment by machack666** [comment]
I would also be concerned about the installation having not completed; if this specific table was missing there are likely other pieces which did not get completely set up.

**Comment by machack666** [comment]
@danieltanfh95 Was this an existing Bucardo installation that was upgraded, by chance?

**Comment by danieltanfh95** [comment]
@machack666 this was on fresh installations, using bucardo 5.4.1, it was tested about a dozen times, all of my installations failed to create that table.

**Comment by machack666** [comment]
@danieltanfh95 so complete teardown/rebuild?

**Comment by danieltanfh95** [comment]
@machack666 yep, complete teardown/rebuild of everything, fresh installs of pgsql 10.

I've done some testing just now again, and so far simply adding the table works, even though the table itself isn't used (iirc)

**Comment by akhon** [comment]
had same issue in ubuntu 14.04/16.04, postgres 9.3/9.5, and bucardo versions 5.4, 5.5, 4.99, fresh machines and installations.

solution mentioned by @danieltanfh95 works, but after creating table manually need to reload config in bucardo via command `bucardo reload config`. reloading service didn't help.
